src/graphics/resources/bind/group.rs

```diff
@@ -0,0 +1,52 @@
+use wgpu::{BindGroup, BindingResource, Buffer, Device};
+
+use crate::graphics::{resources::Texture, Context, ImageSlice};
+
+pub fn source(image: &ImageSlice, ctx: &Context) -> BindGroup {
+    let sampler = create_sampler(&ctx.client.device);
+    let source_texture = Texture::new_source(image, &ctx.client);
+
+    let entries = vec![
+        BindingResource::TextureView(&source_texture.view),
+        BindingResource::Sampler(&sampler),
+    ];
+    create_bind_group("Source", entries, ctx)
+}
+
+pub fn transform(transform_buffer: Buffer, ctx: &Context) -> BindGroup {
+    let entries = vec![transform_buffer.as_entire_binding()];
+    create_bind_group("Source", entries, ctx)
+}
+
+fn create_sampler(device: &Device) -> wgpu::Sampler {
+    use wgpu::{AddressMode, FilterMode};
+
+    device.create_sampler(&wgpu::SamplerDescriptor {
+        address_mode_u: AddressMode::ClampToEdge,
+        address_mode_v: AddressMode::ClampToEdge,
+        mag_filter: FilterMode::Linear,
+        min_filter: FilterMode::Linear,
+        ..Default::default()
+    })
+}
+
+fn create_bind_group(key: &str, entries: Vec<BindingResource>, ctx: &Context) -> BindGroup {
+    use wgpu::BindGroupEntry;
```

Comment:
Why not above with the rest of the includes?

Reply:
The rule of thumb I try to follow is this:

I import a token at the top when it's used more than once. If it's unique to the function, I import it locally.

This is not very important but I can change it if you don't like it.

Reply:
I won't ask you to change it because it's mostly your code and we all have our own ways of doing things... but I do have an opinion in this matter.

- I'm not 100% sure, but this goes against Rust "way of doing things"™ Or to put it differently, I never saw anyone doing this in the Rust community.
- Unless I'm missing something, there's no advantage. At least not for the programmer reading the code and not for the compiler
- However, I do see a disadvantage. When a project gets big, the programmers (the future you, me, etc.) will forget if a token is imported or not, and will import it again. So there will be several imports of the same token in various functions. Unless the Rust-Analyzer auto-completion does it automatically, but I doubt it does.

Imo, leaving the imports at the top of the file where the vast majority of the programmers goes to find them is the "least surprising way".